Readme.md

```diff
@@ -43,24 +43,44 @@ For example: `nix build --arg buildInstaller false --arg buildBundle false` will
 
 A virtual machine (with test instrumentation) can be started without any of the above builds.
 
+## Testing
+
 ### Virtual machine
 
-A helper is available to quickly start a virtual machine:
+Most changes to system configuration and/or the contoller can be tested in a virtual machine.
+To create and run a VM, run:
 
-```
+```bash
 ./build vm
+./result/bin/run-in-vm
 ```
 
 In order to get the vm system journal, look at the output of `run-in-vm`
 for a command starting with `socat`.
 
-See the output of `run-in-vm --help` for more information.
+See the output of `./result/bin/run-in-vm --help` for more information.
 
 #### Guest networking
 
 The default user-mode network stack is used to create a virtual Ethernet connection with bridged Internet access for the guest. If you find that the guest has a dysfunctional Internet connection, check your host's firewall settings. If using ConnMan, restart ConnMan service and try again.
 
-## Testing
+### Testing on PlayOS hardware
+
+Changes such as NixOS upgrades, or to anything else that directly interacts with system hardware may necessitate testing on an actual machine. To do so, build a live system:
```

Comment:
Two comments:

- There is a third possibility that sits between testing on QEMU and a physical machine: A [VM like VirtualBox](https://github.com/dividat/playos/tree/develop/docs/arch#installation-on-virtualbox), which can simulate a system more fully, including the installer, Grub and A/B partitions.
- The live system is very useful, but sometimes it's necessary to install a system.

Reply:
Mentioned these points in:

[Amend readme with additional details](https://github.com/dividat/playos/pull/136/commits/03340e1a0b37b7435071057191b02f382feebee4)

It isn't clearly explained when to use a live system vs a complete installation, because it's not really clear to me 😅. Not sure how important it is to go into such details. I would think the nature of a particular thing being tested would draw us to one of the options.

Reply:
> Not sure how important it is to go into such details. I would think the nature of a particular thing being tested would draw us to one of the options.

I agree, I wanted to point these out because the added instructions initially were very specifically saying to use a live system, which could be misguiding at times.

> It isn't clearly explained when to use a live system vs a complete installation, because it's not really clear to me

As you say, an exhaustive characterization might be hard, but there are some obvious cases, based around the fact that the live system has no A/B partitions and entirely ephemeral storage. So:

- Testing anything to do with the installer requires a full system
- Testing anything to do with A/B systems requires a full system
- Testing anything that spans past one boot (e.g. "Full HD" settings) requires a full system

Reply:
3d2526d